prov/shm/src/smr_util.c

```diff
@@ -573,15 +573,15 @@ void smr_map_del(struct smr_map *map, int64_t id)
 {
 	struct dlist_entry *entry;
 
-	if (id >= SMR_MAX_PEERS || id < 0 || map->peers[id].peer.id < 0)
-		return;
-
```

Comment:
Same comment here, how often do we return from here?  Do not remove if we return from here often.

Reply:
This one I don't think is ever hit. We should never be deleting an id that isn't valid. But again, it's moved to below so it's still there, just after the ep list check for locking simplicity.